pkg/scheduler/api/compatibility/compatibility_test.go

```diff
@@ -1127,128 +1079,134 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
+				"CheckNodeMemoryPressure",
+				"CheckNodeDiskPressure",
+				"CheckNodePIDPressure",
+				"CheckNodeCondition",
+				"MaxEBSVolumeCount",
+				"MaxGCEPDVolumeCount",
+				"MaxAzureDiskVolumeCount",
+				"MaxCSIVolumeCountPred",
+				"MaxCinderVolumeCount",
+				"MatchInterPodAffinity",
+				"GeneralPredicates",
+				"CheckVolumeBinding",
+				"TestServiceAffinity",
+				"TestLabelsPresence",
+			),
+			wantPrioritizers: sets.NewString(
+				"EqualPriority",
+				"ImageLocalityPriority",
+				"LeastRequestedPriority",
+				"BalancedResourceAllocation",
+				"SelectorSpreadPriority",
+				"NodePreferAvoidPodsPriority",
+				"NodeAffinityPriority",
+				"TaintTolerationPriority",
+				"InterPodAffinityPriority",
+				"MostRequestedPriority",
+				"RequestedToCapacityRatioPriority",
+			),
+			wantExtenders: []schedulerapi.ExtenderConfig{{
+				URLPrefix:        "/prefix",
+				FilterVerb:       "filter",
+				PrioritizeVerb:   "prioritize",
+				Weight:           1,
+				BindVerb:         "bind", // 1.11 restored case-sensitivity, but allowed either "BindVerb" or "bindVerb"
+				EnableHTTPS:      true,
+				TLSConfig:        &schedulerapi.ExtenderTLSConfig{Insecure: true},
+				HTTPTimeout:      1,
+				NodeCacheCapable: true,
+				ManagedResources: []schedulerapi.ExtenderManagedResource{{Name: v1.ResourceName("example.com/foo"), IgnoredByScheduler: true}},
+				Ignorable:        true,
+			}},
 		},
 	}
-
 	registeredPredicates := sets.NewString(factory.ListRegisteredFitPredicates()...)
 	registeredPriorities := sets.NewString(factory.ListRegisteredPriorityFunctions()...)
 	seenPredicates := sets.NewString()
 	seenPriorities := sets.NewString()
 
 	for v, tc := range schedulerFiles {
-		fmt.Printf("%s: Testing scheduler config\n", v)
-
-		policy := schedulerapi.Policy{}
-		if err := runtime.DecodeInto(latestschedulerapi.Codec, []byte(tc.JSON), &policy); err != nil {
-			t.Errorf("%s: Error decoding: %v", v, err)
-			continue
-		}
-		for _, predicate := range policy.Predicates {
-			seenPredicates.Insert(predicate.Name)
-		}
-		for _, priority := range policy.Priorities {
-			seenPriorities.Insert(priority.Name)
-		}
-		if !reflect.DeepEqual(policy, tc.ExpectedPolicy) {
-			t.Errorf("%s: Expected:\n\t%#v\nGot:\n\t%#v", v, tc.ExpectedPolicy, policy)
-		}
+		t.Run(v, func(t *testing.T) {
+			policyConfigMap := v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{Namespace: metav1.NamespaceSystem, Name: "scheduler-custom-policy-config"},
+				Data:       map[string]string{schedulerconfig.SchedulerPolicyConfigMapKey: tc.JSON},
+			}
+			client := fake.NewSimpleClientset(&policyConfigMap)
+			algorithmSrc := schedulerconfig.SchedulerAlgorithmSource{
+				Policy: &schedulerconfig.SchedulerPolicySource{
+					ConfigMap: &kubeschedulerconfig.SchedulerPolicyConfigMapSource{
+						Namespace: policyConfigMap.Namespace,
+						Name:      policyConfigMap.Name,
+					},
+				},
+			}
+			informerFactory := informers.NewSharedInformerFactory(client, 0)
```

Comment:
mandatoryPredicates := sets.NewString("CheckNodeCondition"), and take the union of mandatory and want when comparing with schedPredicates.

note that we have to keep "CheckNodeCondition" in the wantPredicates set of the releases that actually introduced it.

Reply:
That's definitely better. Resolved.